reports/sql/report.class.php

```diff
@@ -129,7 +129,10 @@ public function create_report() {
 
             $sql = $this->prepare_sql($sql);
 
-            if (isset($this->filterform) && $this->filterform->get_data() && $rs = $this->execute_query($sql)) {
+            $filterformhasdata = isset($this->filterform) && $this->filterform->get_data();
+            $nofilterform = !isset($this->filterform);
```

Comment:
I find this a little hard to read, purely because its inversed.

Could it maybe be
```
$filterform = isset($this->filterform);
$formhasdata = $filterform && $this->filterform->get_data();

if (($formhasdata || !$filterform) && $rs...)) {
```

There is likely a cleaner way to do this using conditional access with ?-> But thats only since PHP 8.

Reply:
That looks a lot better. This has been updated in the latest commit. Should be ready to squash and merge now